src/lib/data/bluetooth.ts

```diff
@@ -79,58 +58,63 @@ export async function onPowerSensorEvent(
 }
 
 async function connectToSensor(
-  sensor: Sensor | undefined,
-  bleSensor: any,
+  sensorModel: SensorModel,
+  sensor: any,
   sensorType: string,
   callback: {
     (r: RealtimeDataModel): Promise<RealtimeDataModel>;
     (r: RealtimeDataModel): Promise<RealtimeDataModel>;
   },
 ) {
-  bleSensor = getSensorFromType(sensorType);
-  if (sensor === undefined) {
+  if (sensorModel === undefined) {
     throw new Error(`No ${sensorType} sensor found`);
   }
 
   try {
     // Try connect to our BLE sensor
-    bleSensor.address = sensor.address;
-    await bleSensor.connect();
-
+    sensor.address = sensorModel.address;
+    await sensor.connect();
   } catch (error) {
     console.log('catch in connectToSensor was called');
-    await bleSensor.disconnect().catch((err: any) => console.log(err));
+    await sensor.disconnect().catch((err: any) => console.log(err));
     throw error;
   }
   // Start subscription on our sensor
-  bleSensor.subscribe(callback);
-
+  sensor.subscribe(callback);
 }
 
 export function bleSensorService(
-  sensorType: string,
+  sensorType: SensorType,
   callback: (r: RealtimeDataModel) => Promise<RealtimeDataModel>,
 ) {
-  let bleSensor: any = null;
+  // TODO get types for react-native-cylcing-sensors
   let sensor: any = null;
   let observeable: Subscription;
 
   const start = async () => {
     console.log(`Starting ${sensorType} service worker.`);
-
-    let sensors = await getAllSensors();
-    sensor = findFirstSensorOfType(sensors, sensorType);
-
-    const subscription = db.get<SensorModel>('sensor')?.query().observe();
+    sensor = getSensorFromType(sensorType);
+    // get the first sensor of sensorType
+    const subscription = db
+      .get<SensorModel>('sensor')
+      ?.query(
+        Q.unsafeSqlQuery(
+          "select * from sensor where EXISTS (SELECT 1 FROM json_each(sensor_type) WHERE value = ?) and _status is not 'deleted' order by created_at DESC limit 1",
```

Comment:
it's likely that we don't have JSON functions available, as Android's SQLite version is often below 3.9.0. which is where they were introduced :/

Reply:
> This all looks good but we can't use json functions in android SQLite it seems, any other ideas to do it? could potentially move away from json and have a separate "services" table with a key to the sensor?

Yip, a mapping table for services would work. But because [we control the naming of the sensor types](https://github.com/shaunmulligan/react-native-cycling-sensors/blob/main/src/bleSensors.ts#L25-L32) we could just do a `LIKE %<sensor_type>%` we'd just have to be careful not to create any overlapping sensor_types like `HeartRateMonitor` and HeartRateMonitor123`.

Reply:
pushed a change to use `LIKE` and it all works great, so will merge this into my PR